handler/src/main/java/io/netty/handler/ssl/OpenSslX509KeyManagerFactory.java

```diff
@@ -158,61 +158,65 @@ OpenSslKeyMaterialProvider newProvider() {
                 return new OpenSslPopulatedKeyMaterialProvider(keyManager,
                         password, aliases);
             }
+        }
+    }
 
-            /**
-             * {@link OpenSslKeyMaterialProvider} implementation that pre-compute the {@link OpenSslKeyMaterial} for
-             * all aliases.
-             */
-            private static final class OpenSslPopulatedKeyMaterialProvider extends OpenSslKeyMaterialProvider {
-                private final Map<String, Object> materialMap;
-
-                OpenSslPopulatedKeyMaterialProvider(
-                        X509KeyManager keyManager, String password, Iterable<String> aliases) {
-                    super(keyManager, password);
-                    materialMap = new HashMap<String, Object>();
-                    boolean initComplete = false;
-                    try {
-                        for (String alias: aliases) {
-                            if (alias != null && !materialMap.containsKey(alias)) {
-                                try {
-                                    materialMap.put(alias, super.chooseKeyMaterial(
-                                            UnpooledByteBufAllocator.DEFAULT, alias));
-                                } catch (Exception e) {
-                                    // Just store the exception and rethrow it when we try to choose the keymaterial
-                                    // for this alias later on.
-                                    materialMap.put(alias, e);
-                                }
-                            }
-                        }
-                        initComplete = true;
-                    } finally {
-                        if (!initComplete) {
-                            destroy();
+    /**
+     * {@link OpenSslKeyMaterialProvider} implementation that pre-compute the {@link OpenSslKeyMaterial} for
+     * all aliases.
+     */
+    // Only package-private for testing.
+    static final class OpenSslPopulatedKeyMaterialProvider extends OpenSslKeyMaterialProvider {
```

Comment:
all the changes here are just because I moved it to the outer-class and made it package-private for testing.